python/lsst/pipe/base/task_logging.py

```diff
@@ -0,0 +1,244 @@
+# This file is part of task_base.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ("LOG_TRACE",)
+
+import logging
+from logging import LoggerAdapter
+from deprecated.sphinx import deprecated
+from contextlib import contextmanager
+
+try:
+    import lsst.log as lsstLog
+except ModuleNotFoundError:
+    lsstLog = None
+
+# log level for trace (verbose debug)
+LOG_TRACE = 5
+logging.addLevelName(LOG_TRACE, "TRACE")
+
+LOG_VERBOSE = (logging.INFO + logging.DEBUG) // 2
+logging.addLevelName(LOG_VERBOSE, "VERBOSE")
+
+
+class _F:
+    """
+    Format, supporting str.format() syntax
+    see using-custom-message-objects in logging-cookbook.html
+    """
+    def __init__(self, fmt, /, *args, **kwargs):
+        self.fmt = fmt
+        self.args = args
+        self.kwargs = kwargs
+
+    def __str__(self):
+        return self.fmt.format(*self.args, **self.kwargs)
+
+
+class TaskLogAdapter(LoggerAdapter):
+    """A special logging adapter to provide log features for `Task`.
+
+    This is easier to use for a unified interface to task logging
+    than calling `logging.setLoggerClass`.  The main simplification
+    is that this will work with the root logger as well as loggers
+    created for tasks.
+    """
+
+    @contextmanager
+    def temporary_log_level(self, level):
+        """A context manager that temporarily sets the level of this logger.
+
+        Parameters
+        ----------
+        level : `int`
+            The new temporary log level.
+        """
+        old = self.level
+        self.setLevel(level)
+        try:
+            yield
+        finally:
+            self.setLevel(old)
+
+    @property
+    def level(self):
+        return self.logger.level
+
+    @property
+    def TRACE(self):
+        return LOG_TRACE
+
+    @property
+    def VERBOSE(self):
+        return LOG_VERBOSE
+
+    @property
+    @deprecated(reason="Use logging.DEBUG. Will be removed after v23.",
```

Comment:
If we're going to need to use `logger.VERBOSE`, I think we should continue to allow `logger.DEBUG`.

Reply:
I've changed all the deprecations if you would like to take a look. I now am allowing all the constants to exist without deprecation.  I've also made TaskLogAdapter public (mostly so it gets documented in the sphinx docs) but I'm okay with reversing that and instead adding an explicit line in the sphinx.